Editor Setup page w VS Code Extension info (#279)
withastro/docs: src/pages/en/editor-setup.md

---
layout: ~/layouts/MainLayout.astro
title: Editor Setup
description: Set up your editor to build with Astro.
---

> You can build your Astro site with any code editor, installed locally or in the cloud!
> 
> [VS Code](https://code.visualstudio.com) is a popular code editor that can be dowloaded locally, and is also used by online platforms such as [GitHub.dev](https://github.dev), [GitHub Codespaces](https://github.com/features/codespaces) and [Gitpod](https://gitpod.io).
> 
> 
> ## VS Code Extension

Reviewer: Might it make sense to move the second paragraph below the VS Code header? Something like 

```suggestion
You can build your Astro site with any code editor, installed locally or in the cloud!

This page explains how to improve the experience of editing `.astro` files in supported editors.

## VS Code Extension

[VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded locally, and is also used by online platforms such as [GitHub.dev](https://github.dev), [GitHub Codespaces](https://github.com/features/codespaces) and [Gitpod](https://gitpod.io).
```

Author: Will do!

> Astro has its own [Astro VS Code Extension on Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode) that you can install in any of these VS Code environments for .`astro` syntax highligting, code completion, and more!

Reviewer 2: In web environments, currently the only supported feature is the syntax highlighting :sweat_smile:

Everything else is locked behind https://github.com/withastro/language-tools/issues/180

Author: Yes, I think that's all I get in CodeSandbox, so I've specifically said only syntax highlighting below in the online editors setion.  Gitpod gives me the full extension. (After almost a year coding in monochrome, without even syntax highlighting, when it started autocompleting stuff on me, it was quite disorienting!)

Author: Yup, Gitpod gives me the full version, but github.dev only gives me the syntax highlighting. (I'm assuming that's "VS Code Browser" instead of full "VS Code.")  So, I'll reorganize/update to make things accurate.

### Full List of Features
- [Go to Definition](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition)
- Code hover hints 
- Code completion
- Function signatures
- Syntax highlighting
- Code folding
- Emmet

After [setting up VS Code](https://code.visualstudio.com/Docs/setup/setup-overview), to get started building with Astro:

1. Navigate to the "Extensions" tab ("Ctrl-Shift-X")
2. Search for "astro" in the VS Code Marketplace
3. Install the [Astro VS Code Extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode)

## Other Astro language extensions

You can also get Astro language support for other code editors using:

> [Astro VS Code Extension on Open VSX](https://open-vsx.org/extension/astro-build/astro-vscode) - the complete VS Code Extension, available on the Open VSX registry for open platforms like Gitpod.

Reviewer 2: A lot of users initially asked for this because they use "free" versions of VS Code such as VSCodium, it'd be cool to mention those in addition to GitPod

Author: Absolutely!

> [Nova editor extension](https://extensions.panic.com/extensions/sciencefidelity/sciencefidelity.astro/) - Syntax highlighting, IntelliSense, autocompletion for 🚀 Astro

Reviewer 2: I think it should be mentioned that only the VS Code integration is official at the moment

Author: Great. I'll note that Nova is a "community contribution" or something like that.

---
layout: ~/layouts/MainLayout.astro
setup: |
  import Badge from '~/components/Badge.astro';
title: Editor Setup
description: Set up your editor to build with Astro.
---

> You can build your Astro site with any code editor, installed locally or in the cloud!

Reviewer 3: Pizzaz people!! 

```suggestion
You can always build and develop your Astro project using you any off your favorite code editors, either locally or on the cloud!
```

Author: ```suggestion
You can build and develop your Astro project using any of your favorite code editors, either locally or in the cloud!
```

> This page explains how to improve the experience of editing `.astro` files in supported editors.

Reviewer 3: ```suggestion
On this page, we demonstrate how to improve the overall experience of working with `.astro` files in supported editors.
```

Author: ```suggestion
See how to improve the experience of working with `.astro` files in supported editors.
```

> Improve your experience by working with `.astro` files in supported editors with Astro language support.

Reviewer: Too much support?

```suggestion
Improve your experience by working with `.astro` files in editors with Astro language support.
```

Reviewer: (Sorry missed the fresh eyes/later memo 😬)

Author: Haha, is there EVER too much support??  :)

Fred's pushing some changes too, so we'll wait on that, at least.

## VS Code Extension

> [VS Code](https://code.visualstudio.com) is a popular code editor that can be dowloaded locally, and is also used by online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io). 

Reviewer: Typo:

```suggestion
[VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded locally, and is also used by online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io). 
```

Author: Fixed!

> [VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded locally, and is also used by online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io). 

Reviewer 3: ```suggestion
[VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded and operated locally as well as on online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io). 
```

Author: OK, I'm gonna give you one!!! Enjoy this!

(But, I'm gonna change operated to used.)

Author: ```suggestion
[VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded and used locally as well as on online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io).
```

> [VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded and used locally as well as on online platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io).

Author: ```suggestion
[VS Code](https://code.visualstudio.com) is a popular code editor that can be downloaded and used locally as well as in full online development platforms such as [GitHub Codespaces](https://github.com/features/codespaces), and [Gitpod](https://gitpod.io).
```

> Astro has its own [Astro VS Code Extension on Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode) that you can install in any of these VS Code environments for .`astro` syntax highligting, code completion, and more!

Author: ```suggestion
Astro has its own [Astro VS Code Extension] (https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode) that you can install in any of these VS Code environments for .`astro` syntax highligting, code completion, and more!
```

### Full List of Features

> - [Go to Definition](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition)

Reviewer 3: Either all these get links or they dont, 
```suggestion
- [Go to Definition](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition)
```

Author: I did this because in our official extension page, it has a link (and only this one). I've also seen it (and only it) linked in a feature set elsewhere. So, I don't know if this is a new thing, or it's special or what. But, that's what we do in the VS Marketplace/Open VSX registry, so there's precedent.

Reviewer 2: To be honest it's a bit of a legacy list from the time where the features supported were really precise. Nowadays, we have *almost* (I'm working on it!) all the "basic" features like these and the only features we would really need to list are those that are Astro specific

- Code hover hints 
- Code completion
- Function signatures
- Syntax highlighting
- Code folding
- Emmet

> 

Reviewer 3: There is more to this list, but man I've just drawn a blank, It can be added to in the future

Author: Yes, anyone is welcome to add to it in the future! I've just copied from our offical extension description.

### Installation

> After [setting up VS Code](https://code.visualstudio.com/Docs/setup/setup-overview) and running any of our [Astro starter templates](https://github.com/withastro/astro/tree/main/examples), the Astro VS Code extension should be recommended to you automatically by a pop-up (in the bottom right corner).

Reviewer 3: ```suggestion
Deploying your Astro Project using any of our [Astro starter templates](https://github.com/withastro/astro/tree/main/examples), from within VS Code. VS Code  should prompt you to install the Astro VS Extension to you automatically via a pop-up in the bottom right corner.
```

> To add the extension yourself:

Reviewer 3: ```suggestion
### Manual Installation
```

Author: I think this section is so short, it doesn't warrant another heading. (And, don't think it needs to show up in the sidebar.)

> 1. Navigate to the "Extensions" tab ("Ctrl-Shift-X")

Reviewer: If we want to include the keyboard shortcut it should probably include the Mac variation (Cmd+Shift+X).

Also, there’s a special `<kbd>` element for keyboard shortcuts if you fancy using it:

```html
<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>X</kbd>
```

Renders:

<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>X</kbd>

(Not sure if the docs site has styling in place for `<kbd>`.)

Author: Doesn't appear to...  :(  Maybe we need to bug @bholmesdev about this....

![Screenshot 2022-03-29 17 53 01](https://user-images.githubusercontent.com/5098874/160705194-f89670ef-cc01-4990-adbc-02caf6662ed0.png)

So, maybe for both reasons (no fancy styling, needing to type multiple variations), I'll just take it out. Another site gave the keyboard shortcuts for their install instructions, and I thought that was kind of neat as visual variety. But I don't want to make it a Whole. Thing.

Reviewer 4: You can also link to this docs page on extensions! Should keep our instructions evergreen too: https://code.visualstudio.com/docs/editor/extension-marketplace#_browse-for-extensions

Reviewer: I’d happily add some custom styles, but I also think the keyboard shortcut may be superfluous in this case.

2. Search for "astro" in the VS Code Marketplace
3. Install the [Astro VS Code Extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode)

> ## Other Astro language extensions

Reviewer 3: ```suggestion
## Alternative Astro Language Extensions
```

Author: ```suggestion
## Other Astro Language Extensions
```

> You can also get Astro language support using:

Reviewer 3: ```suggestion
Astro has found itself being used in other alternative code environments to VS Code, here we provide additional information on where else you can use and experience the same set of features.
```

Author: ```suggestion
You can also get Astro language support outside of VS Code using:
```
